### PR TITLE
FM2-205: Update medication resource to support paging

### PR DESCRIPTION
**Pull request by varung-31**
## Description of what I changed

Update medication resource to return paged search results
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-205

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/fhir2/api/search/MedicationSearchQueryImplTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		IBundleProvider result = search(theParams);
+
+		List<IBaseResource> resultList = get(result);
+
+		assertThat(result, notNullValue());
+		assertThat(resultList.size(), equalTo(0));
+	}
+
+	@Test
+	public void searchForMedications_shouldSearchMedicationsByIngredientCode() {
```
Comment (varung-31) [inline api/src/test/java/org/openmrs/module/fhir2/api/search/MedicationSearchQueryImplTest.java:134]: This is the test which gives me the ambiguous column name error.

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationDaoImpl.java [hunk 2]
```diff
@@ -34,18 +34,36 @@ public Drug createOrUpdate(Drug drug) {
...
 	}
-	
+
+	private void handleIngredientCode(Criteria criteria, TokenAndListParam ingredientCode) {
+		if (ingredientCode != null) {
+			criteria.createAlias("ingredients", "i");
+			criteria.createAlias("i.ingredient", "ic");
+
+			handleCodeableConcept(criteria, ingredientCode, "ic", "icm", "icrt").ifPresent(criteria::add);
+		}
```
[the remark this answers is not in the document]
Comment (varung-31) [reply api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationDaoImpl.java:64]: Seeing the query logs, the concept table is joining twice and that's why it is not able to distinguish between the column names. But why it is joining twice is still a matter of concern.

[the remark this answers is not in the document]
Comment (varung-31) [reply api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationDaoImpl.java:72]: Resolved using DetachedCriteria.

**Comment by varung-31** [comment]
@corneliouzbett  Can you also please take a look at this?

**Comment by corneliouzbett** [comment]
@varung-31 can you resolve the conflicts

**Comment by varung-31** [comment]
> @varung-31 can you resolve the conflicts

@corneliouzbett  Done

**Pull request merged**